app/vmselect/promql/exec.go

```diff
@@ -26,8 +26,28 @@ var (
 		`This option is DEPRECATED in favor of {__graphite__="a.*.c"} syntax for selecting metrics matching the given Graphite metrics filter`)
 )
 
+// ExecErr is an error type returned from Exec fn.
+// This type is can be checked by the caller in order
+// to modify the error when needed.
+type ExecErr struct {
```

Comment:
I'm unsure this type is needed here. It may be more clearer and more maintainable to just wrap the innermost error, which must be returned to the user, into a similar type (I'd name it `UserReadableError`), and then just use the following code for obtaining this error just before returning the result to the client:

```go
var urr UserReadableError
if errors.As(err, &urr) {
  // return urr to the client
} else {
  // return err to the client
}
```

Reply:
Thanks! Please see code, commit and PR description updated accordingly.